### PR TITLE
Sqrt rsqrt

### PR DESCRIPTION
**Pull request by jtlap**

### CODE REVIEW

File: include/eve/module/core/regular/impl/simd/x86/rsqrt.hpp [hunk 1]
```diff
@@ -32,151 +32,156 @@ namespace eve::detail
...
+        (std::is_same_v<v_t, double> && eve::any(eve::abs(x) < smallestposval(eve::as<float>()) ||
+                                                 eve::abs(x) > valmax(eve::as<float>()))) )
+      // this is necessary because of the poor initialisation by float intrinsic
+    {
+      auto [a00, nn] = ifrexp[pedantic](x);
+      auto tst       = is_odd(nn);
+      nn             = dec[tst](nn);
+      a00            = mul[tst](a00, 2);
+      auto a0        = rsqrt_x86_normal(a00);
+      std::cout << ldexp[pedantic](a0, -nn / 2) << std::endl;
```
Comment (jfalcou) [inline include/eve/module/core/regular/impl/simd/x86/rsqrt.hpp:78]: Remove cout

File: include/eve/module/core/regular/impl/simd/x86/rsqrt.hpp [hunk 2]
```diff
@@ -32,151 +32,156 @@ namespace eve::detail
...
-    return if_else(is_eqz(x), inf(eve::as(x)), ldexp[pedantic](a0, -nn / 2));
+    if constexpr(O::contains(raw2))
+    {
+      if constexpr( std::is_same_v<T, double> )
+      {
+        // The maximum error for this approximation is 1.5e-12
+        return _mm_cvtps_pd(_mm_rsqrt_ps(_mm_cvtpd_ps(a0)));
+      }
+      else if constexpr( std::is_same_v<T, float> )
+      { return _mm_rsqrt_ps(a0); }
```
Comment (jfalcou) [inline include/eve/module/core/regular/impl/simd/x86/rsqrt.hpp:104]: skip lines or remove the { }  and put on previosu line

File: include/eve/module/core/regular/rsqrt.hpp [hunk 3]
```diff
@@ -7,11 +7,20 @@
...
-#include <eve/arch.hpp>
-#include <eve/detail/overload.hpp>
 
 namespace eve
 {
+  template<typename Options>
+  struct rsqrt_t : elementwise_callable<rsqrt_t, Options, raw_option, pedantic_option>
+  {
+    template<eve::floating_value T>
+    requires(eve::same_lanes_or_scalar<T>)
```
Comment (jfalcou) [inline include/eve/module/core/regular/rsqrt.hpp:17]: useless, there is only one parameter

File: include/eve/module/core/regular/sqrt.hpp [hunk 4]
```diff
@@ -8,12 +8,22 @@
...
+#include <eve/traits/overload.hpp>
+#include <eve/module/core/decorator/core.hpp>
 
 namespace eve
 {
+  template<typename Options>
+  struct sqrt_t : elementwise_callable<sqrt_t, Options, raw_option>
+  {
+    template<eve::value T>
+    requires(eve::same_lanes_or_scalar<T>)
```
Comment (jfalcou) [inline include/eve/module/core/regular/sqrt.hpp:20]: useless, there is only one parameter

**Review by jfalcou — CHANGES_REQUESTED** [review]

File: include/eve/module/core/regular/impl/simd/x86/rsqrt.hpp (hunk 2 above)
Comment (jtlap) [reply include/eve/module/core/regular/impl/simd/x86/rsqrt.hpp:104]: done

File: include/eve/module/core/regular/rsqrt.hpp (hunk 3 above)
Comment (jtlap) [reply include/eve/module/core/regular/rsqrt.hpp:17]: yes !

File: include/eve/module/core/regular/impl/simd/x86/rsqrt.hpp (hunk 1 above)
Comment (jtlap) [reply include/eve/module/core/regular/impl/simd/x86/rsqrt.hpp:78]: done

File: include/eve/module/core/regular/rsqrt.hpp [hunk 5]
```diff
@@ -7,11 +7,19 @@
 //==================================================================================================
 #pragma once
 
-#include <eve/arch.hpp>
```
Comment (jfalcou) [inline include/eve/module/core/regular/rsqrt.hpp:10]: this lacks
```
#include <eve/arch.hpp>
#include <eve/traits/overload.hpp>
#include <eve/module/core/decorator/core.hpp>
```

Comment (jtlap) [reply include/eve/module/core/regular/rsqrt.hpp:10]: done

**Review by jfalcou — APPROVED** [review]

**Review by jfalcou — APPROVED** [review]

**Pull request merged**